### ISSUE TITLE
Noisy trace on SIGINT to vagrant up

### ISSUE THREAD

**Issue by mcandre** [hashicorp/vagrant#10603]
### Vagrant version

Vagrant 2.2.2

### Host operating system

macOS Mojave

### Guest operating system

Debian Buster

### Expected behavior

Vagrant responds quietly to `Ctrl+C`.

### Actual behavior

Vagrant emits a long trace that complains about a synchronize method attempting (and failing) to run in a trap context:

```console
$ vagrant up
<Ctrl+C>
/opt/vagrant/embedded/gems/2.2.2/gems/concurrent-ruby-1.1.3/lib/concurrent/collection/map/mri_map_backend.rb:18:in `synchronize': can't be called from trap context (ThreadError)
	from /opt/vagrant/embedded/gems/2.2.2/gems/concurrent-ruby-1.1.3/lib/concurrent/collection/map/mri_map_backend.rb:18:in `[]='
	from /opt/vagrant/embedded/gems/2.2.2/gems/i18n-1.1.1/lib/i18n.rb:358:in `normalize_key'
	from /opt/vagrant/embedded/gems/2.2.2/gems/i18n-1.1.1/lib/i18n.rb:298:in `normalize_keys'
	from /opt/vagrant/embedded/gems/2.2.2/gems/i18n-1.1.1/lib/i18n/backend/simple.rb:84:in `lookup'
	from /opt/vagrant/embedded/gems/2.2.2/gems/i18n-1.1.1/lib/i18n/backend/base.rb:30:in `translate'
	from /opt/vagrant/embedded/gems/2.2.2/gems/i18n-1.1.1/lib/i18n.rb:185:in `block in translate'
	from /opt/vagrant/embedded/gems/2.2.2/gems/i18n-1.1.1/lib/i18n.rb:181:in `catch'
	from /opt/vagrant/embedded/gems/2.2.2/gems/i18n-1.1.1/lib/i18n.rb:181:in `translate'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/action/runner.rb:59:in `block in run'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/util/busy.rb:49:in `block in fire_callbacks'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/util/busy.rb:49:in `each'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/util/busy.rb:49:in `fire_callbacks'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/util/busy.rb:33:in `block (2 levels) in register'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/batch_action.rb:127:in `join'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/batch_action.rb:127:in `block in run'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/batch_action.rb:65:in `each'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/batch_action.rb:65:in `run'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/environment.rb:280:in `block (2 levels) in batch'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/environment.rb:275:in `tap'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/environment.rb:275:in `block in batch'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/environment.rb:274:in `synchronize'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/environment.rb:274:in `batch'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/plugins/commands/up/command.rb:97:in `execute'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/cli.rb:58:in `execute'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/lib/vagrant/environment.rb:291:in `cli'
	from /opt/vagrant/embedded/gems/2.2.2/gems/vagrant-2.2.2/bin/vagrant:164:in `<main>'
```

### Steps to reproduce

1. Begin a `vagrant up` of a guest with a long download time, say multiple GB over a few MB/sec connection.
3. Interrupt the box download partway with `Ctrl+C`.

**Comment by Boycce** [comment]
For me i can't SSH into the box until i Ctrl+C which shows the same error

> opt/vagrant/embedded/gems/2.2.3/gems/concurrent-ruby-1.1.4/lib/concurrent/collection/map/mri_map_backend.rb:18:in `synchronize': can't be called from trap context (ThreadError)

**Comment by piotrku** [comment]
Same here, any solutions?

**Comment by briancain** [comment]
Hi, yes, unfortunately this was introduced when we upgraded the i18n dependency. We will have a fix soon, thanks!

**Comment by Sennewald** [comment]
Same error here:

ringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'ubuntu/xenial64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'ubuntu/xenial64' version '20170525.0.0' is up to date...
==> default: A newer version of the box 'ubuntu/xenial64' for provider 'virtualbox' is
==> default: available! You currently have version '20170525.0.0'. The latest is version
==> default: '20190708.0.0'. Run `vagrant box update` to update.
==> default: Setting the name of the VM: www-optimal-systems-de_default_1563373213292_58554
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 80 (guest) => 8080 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: ubuntu
    default: SSH auth method: password
C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/concurrent-ruby-1.1.4/lib/concurrent/collection/map/mri_map_backend.rb:18:in `synchronize': can't be called from trap context (ThreadError)
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/concurrent-ruby-1.1.4/lib/concurrent/collection/map/mri_map_backend.rb:18:in `[]='
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/i18n-1.1.1/lib/i18n.rb:358:in `normalize_key'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/i18n-1.1.1/lib/i18n.rb:298:in `normalize_keys'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/i18n-1.1.1/lib/i18n/backend/simple.rb:84:in `lookup'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/i18n-1.1.1/lib/i18n/backend/base.rb:30:in `translate'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/i18n-1.1.1/lib/i18n.rb:185:in `block in translate'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/i18n-1.1.1/lib/i18n.rb:181:in `catch'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/i18n-1.1.1/lib/i18n.rb:181:in `translate'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/action/runner.rb:59:in `block in run'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/util/busy.rb:49:in `block in fire_callbacks'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/util/busy.rb:49:in `each'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/util/busy.rb:49:in `fire_callbacks'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/util/busy.rb:33:in `block (2 levels) in register'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/batch_action.rb:127:in `join'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/batch_action.rb:127:in `block in run'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/batch_action.rb:65:in `each'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/batch_action.rb:65:in `run'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/environment.rb:280:in `block (2 levels) in batch'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/environment.rb:275:in `tap'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/environment.rb:275:in `block in batch'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/environment.rb:274:in `synchronize'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/environment.rb:274:in `batch'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/plugins/commands/up/command.rb:97:in `execute'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/cli.rb:58:in `execute'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/environment.rb:291:in `cli'
        from C:/HashiCorp/Vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/bin/vagrant:182:in `<main>'
